vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/ad_targeting.cc

```
@@ -48,4 +48,32 @@ SegmentList AdTargeting::GetSegments() const {
   return segments;
 }
 
+SegmentList AdTargeting::GetInterestSegments() const {
+  SegmentList segments;
+
+  if (features::IsTextClassificationEnabled()) {
```

nit: should fail early